driver-core/src/main/com/mongodb/operation/ChangeStreamBatchCursor.java

```diff
@@ -139,6 +144,8 @@ public void remove() {
             wrapped.close();
 
             if (resumeToken != null) {
+                changeStreamOperation.startAfter(null);
+                changeStreamOperation.startAtOperationTime(null);
```

Comment:
Also fixing a bug?

Reply:
This resets any user settings - as the resumeToken wins when retrying.